fabcal/forms.py

```diff
@@ -33,25 +33,64 @@ class AbstractSlotForm(forms.Form):
         required=False,
         )
 
+    machine = forms.ModelMultipleChoiceField(
+        queryset = Machine.objects.filter(reservable=True),
+        widget=forms.CheckboxSelectMultiple(
+            attrs={'checked' : ''}
+        ),
+        label=_('Machines'),
+        required=False
+    )
+
     def __init__(self, *args, **kwargs):
         kwargs["label_suffix"] = ""
         super().__init__(*args, **kwargs)
     class Meta:
         abstract = True
 
     def clean_start(self):
+        if 'date' in self.data:
+            start_date = self.data['date']
+        else:
+            start_date = self.data['start_date']
+
         self.cleaned_data['start'] = datetime.datetime.combine(
-            dateparser.parse(self.data['date']), 
+            dateparser.parse(start_date), 
             dateparser.parse(self.data['start_time']).time()
             )
 
+        # check if opening slot already exist / for update
+        if 'opening' in self.initial:
+
+            # check if user alread booked a modified slot
+            qs = MachineSlot.objects.filter(opening_slot=OpeningSlot.objects.get(pk=self.initial['id']))
+
+            for obj in qs:
+                if obj.start < self.cleaned_data['start'] and obj.user:
```

Comment:
Can be directly filtered in the QuerySet

Reply:
#208